.github/workflows/ci_macos.yml

```
@@ -25,15 +25,20 @@ defaults:
 jobs:
   build:
     name: ${{ matrix.compiler }} C++${{ matrix.cpp }} ${{ matrix.build_type }}
-    runs-on: macos-12
+    runs-on: macos-13
```

```suggestion
    runs-on: macos-14
```

macos-13 is x86_64, macos-14 is aarch64/M1